ucm2/Qualcomm/sdm845/HiFi-MM1.conf

```diff
@@ -0,0 +1,78 @@
+# Use case configuration for LenovoYOGAC6301.
+# Author: Srinivas Kandagatla <srinivas.kandagatla@linaro.org>
+
+SectionVerb {
+
+	EnableSequence [
+		cset "name='SLIMBUS_0_RX Audio Mixer MultiMedia1' 1"
+		cset "name='MultiMedia2 Mixer SLIMBUS_0_TX' 1"
+	]
+
+	Include.wcde.File "/codecs/wcd934x/DefaultEnableSeq.conf"
+	Include.wsae.File "/codecs/wsa881x/DefaultEnableSeq.conf"
+
+	Include.wcdd {
+		File "/codecs/wcd934x/DefaultDisableSeq.conf"
+		Before.DisableSequence "0"
+	}
+
+	DisableSequence [
+		cset "name='SLIMBUS_0_RX Audio Mixer MultiMedia1' 0"
+		cset "name='MultiMedia2 Mixer SLIMBUS_0_TX' 0"
+	]
+
+	Value {
+		TQ "HiFi"
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment "Speaker playback"
+
+	Include.wcde.File "/codecs/wcd934x/SpeakerEnableSeq.conf"
+	Include.wsae.File "/codecs/wsa881x/SpeakerEnableSeq.conf"
+	Include.wsad.File "/codecs/wsa881x/SpeakerDisableSeq.conf"
+	Include.wcdd.File "/codecs/wcd934x/SpeakerDisableSeq.conf"
+
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId}"
+		PlaybackVolume "RX2 Digital Volume"
+	}
+}
+
+SectionDevice."Headphones" {
+	Comment "Headphones playback"
+
+	ConflictingDevice [
+		"Speaker"
+	]
+
+	Include.wcdhpe.File "/codecs/wcd934x/HeadphoneEnableSeq.conf"
+	Include.wcdhpd.File "/codecs/wcd934x/HeadphoneDisableSeq.conf"
+
+	Value {
+		PlaybackPriority 200
+		PlaybackPCM "hw:${CardId}"
+		PlaybackVolume "RX1 Digital Volume"
```

Comment:
It's not allowed to define two values with one identifier. The alsa-lib now supports to join two controls into one. Also, the pulseaudio use `PlaybackMixerElem` which refers to the mixer API (not the controls API).

````
LibraryConfig.remap.Config {

	ctl.default.map {
		# Merge two mono controls into one stereo
		"name=''RX Digital Volume" {
			"name='RX1 Digital Volume'".vindex.0 0   # dst (RX) index 0, src index 0 (RX1)
			"name='RX2 Digital Volume'".vindex.1 0   # dst (RX) index 1, src index 0 (RX2)
		}
	}

}
````

So the `SectionDevice` config should be (without `PlaybackVolume` values for the control interface):

````
Value {
  PlaybackPCM "default:${CardId}"
  PlaybackMixerElem "RX Digital"
}
````

The full example (which just renames the controls): https://github.com/alsa-project/alsa-ucm-conf/commit/ddd10778b19be5dc7fb0a0813c4750bac60dfdcb

Also, it would be nice to test the configs with pulseaudio or pipewire.

Reply:
I have tried this, it does not work with my setup, I have alsalib 1.2.4.1 ? Do I need any particular version for this?
I checked amixer after changing volume but I see its not changed at all.

Reply:
The support for the control remap is only in the current alsa-lib repository. I hope to release alsa-lib 1.2.5 very soon.

Reply:
Hi Jaroslav,
I did try this today with the lasted alsa-lib HEAD, I still do not see any change in volume controls 

here is what have in my code,
`LibraryConfig.remap.Config {

	ctl.default.map {
		# Merge two mono controls into one stereo
		"name='HP Digital Volume'" {
			"name='RX1 Digital Volume'".vindex.0 0   # dst (RX) index 0, src index 0 (RX1)
			"name='RX2 Digital Volume'".vindex.1 0   # dst (RX) index 1, src index 0 (RX2)
		}
	}

}`

and in the SectionDevice I use
`Value {
  PlaybackPCM "default:${CardId}"
  PlaybackMixerElem "HP Digital"
}`

pulseaudio complains that
"pulseaudio[1834]: Path HP Digital is not a volume control" 

Do I need to upgrade pulseaudio aswell for this?

--srini

Reply:
Do you have also `Include.card-init` and `Include.ctl-remap` lines from the https://github.com/alsa-project/alsa-ucm-conf/commit/ddd10778b19be5dc7fb0a0813c4750bac60dfdcb in the top level file? The `ctl.default.map` config refines the `/lib/ctl-remap.conf` configuration.

Reply:
Hi Jaroslav, thanks for suggestion, I did indeed missed those includes. But even after adding them to top level conf file as suggested in commit ddd1077 along with update to Syntax 4, I still get same error ""pulseaudio[1834]: Path HP Digital is not a volume control"" 
BTW am still using pulseaudio 14.2, does this matter ?
--srini

Reply:
Some debugging hints:

* `alsaucm -c hw:0 get _alibcfg`    # get the private alsa-lib configuration for first card
* `alsaucm -c hw:0 set _fboot ''`    # this command should create /var/lib/alsa/card0.conf.d/ctl-remap.conf file
* `amixer contents`                        # check, if 'HP Volume' mixer control exists

The `amixer` should  be recent (alsa-utils 1.2.5).

Reply:
I can not see that HP Volume in amixer contents , below is the output from the laptop:

srini@srini:~$ alsaucm -c hw:0 get _alibcfg
  _alibcfg=ctl.hw {
	@args.0 CARD
	@args.CARD.type string
	type hw
	card $CARD
}
ctl.default {
	@args.0 CARD
	@args.CARD.type string
	type remap
	child {
		type hw
		card $CARD
	}
	map {
		"name='HP Digital Volume'" {
			"name='RX1 Digital Volume'".vindex.0 0
			"name='RX2 Digital Volume'".vindex.1 0
		}
	}
}
pcm.hw {
	@args {
		0 CARD
		1 DEV
		2 SUBDEV
		CARD.type string
		DEV.type integer
		SUBDEV.type integer
	}
	type hw
	card $CARD
	device $DEV
	subdevice $SUBDEV
}

srini@srini:~$ alsaucm -c hw:0 set _fboot ''
ALSA lib main.c:778:(execute_sequence) exec '/bin/rm -rf /var/lib/alsa/card0.conf.d' failed (exit code 1)
srini@srini:~$ ls /var/lib/alsa/card0.conf.d/ctl-remap.conf 
/var/lib/alsa/card0.conf.d/ctl-remap.conf
srini@srini:~$ ls /var/lib/alsa/card0.conf.d/ctl-remap.conf  -al
-rw-r--r-- 1 root root 229 Jun  8 10:45 /var/lib/alsa/card0.conf.d/ctl-remap.conf
srini@srini:~$ amixer contents
numid=4,iface=MIXER,name='Master Playback Switch'
  ; type=BOOLEAN,access=rw------,values=1
  : values=on
numid=3,iface=MIXER,name='Master Playback Volume'
  ; type=INTEGER,access=rw------,values=2,min=0,max=65536,step=1
  : values=65536,65536
numid=2,iface=MIXER,name='Capture Switch'
  ; type=BOOLEAN,access=rw------,values=1
  : values=on
numid=1,iface=MIXER,name='Capture Volume'
  ; type=INTEGER,access=rw------,values=2,min=0,max=65536,step=1
  : values=60695,60695
srini@srini:~$ amixer --version
amixer version 1.2.5

Reply:
> srini@srini:~$ amixer contents

Appologize. This command should be `amixer -c 0 contents` . You see rerouted default device to pulseaudio or pipewire.

Reply:
Yes, default device is with pulseaudio. With amixer -c 0 contents I can see the 'HP Digital Volume Mixer' and I can set it manually which changes both RX volumes that side seems to work fine.
However this control appears only after running those two alsaucm commands. I still see Pulseaudio error "Path HP Digital is not a volume control" I get same error even after restarting pulse.

Reply:
Ok, thank you for the clarification. Could you add `PlaybackMixer "default:${CardId}"` line to the `PlaybackMixerElem` line?

Reply:
after adding PlaybackMixer "default:${CardId} like below I get pulse errors like:

"pulseaudio[1886]: Element 'HP Digital' seems to have disappeared. 
"pulseaudio[1886]: Writing HW volume failed.

`
SectionDevice."Headphones" {
        Comment "Headphones playback"

        ConflictingDevice [
                "Speaker"
        ]

        Include.wcdhpe.File "/codecs/wcd934x/HeadphoneEnableSeq.conf"
        Include.wcdhpd.File "/codecs/wcd934x/HeadphoneDisableSeq.conf"

        Value {
                PlaybackPriority 200
                PlaybackPCM "hw:${CardId}"
                PlaybackMixer "default:${CardId}"
                PlaybackMixerElem "HP Digital"
        }
}

`

Reply:
Could you, please, post pulseaudio vebose log with and without the `PlaybackMixer` line (for example via https://gist.github.com)?

(https://fedoraproject.org/wiki/How_to_debug_PulseAudio_problems)

Reply:
Here is the requested output form pulseaudio:

https://gist.github.com/Srinivas-Kandagatla/c9b0d5c9746166463d7d46cc25a8e0ee